src/Brick/Widgets/Core.hs

```diff
@@ -968,18 +968,27 @@ cached n w =
     Widget (hSize w) (vSize w) $ do
         result <- cacheLookup n
         case result of
-            Just prevResult -> return prevResult
+            Just (clickables, prevResult) -> do
+                clickableNamesL %= (clickables ++)
+                return prevResult
             Nothing  -> do
                 wResult <- render w
-                cacheUpdate n wResult
+                clickables <- renderedClickables wResult
```

Comment:
I believe the list comprehension in `renderedClickables` is redundant, and this can be reduced to just `clickables <- use allClickablesL`.

Reply:
Oh, I see why you did this now. You are limiting the clickable names to just the clickable names whose extents were in the rendered result. Please disregard my comment!

Reply:
(Could you add a Haskell comment to this code explaining that nuance?)